### PR TITLE
fix: test and fix types

### PR DESCRIPTION
**Pull request by qwelias**
- added tests for types
- fixed types
- added peek on Computed
- fixed toString to always return string

### CODE REVIEW

File: esm/index.js [hunk 1]
```diff
@@ -46,23 +39,38 @@ export class Signal {
...
+   */
   constructor(_, v, o, f) {
     super(_);
     this.f = f;                   // is effect?
     this.$ = true;                // should update ("value for money")
     this.r = new Set;             // related signals
     this.s = new Reactive(v, o);  // signal
   }
-  /** @readonly */
+  peek() { return this.s.peek() }
```
Comment (WebReflection) [inline esm/index.js:73]: this is inherited ... it shouldn't be needed

**Comment by WebReflection** [comment]
beside conflicts with what's in the main, I am not sure why changes to the code are landing ... it's OK for the `String(this.value)` but the rest looks off, specially added peek() for the computed which should've been covered already and it's inherited like it is in signals

**Comment by WebReflection** [comment]
tests failing now

**Comment by WebReflection** [comment]
I've merged the other ... let's see how this goes but it need a rebase from main

**Comment by ilibar-zpt** [comment]
I have removed all changes besides the types test for now, which need to be fixed

Before i can fix them i need an answer [here](https://github.com/WebReflection/usignal/pull/25#discussion_r1105891068) as i cannot understand it

**Comment by qwelias** [comment]
I have removed all changes besides the types test for now, which need to be fixed

Before i can fix them i need an answer [here](https://github.com/WebReflection/usignal/pull/25#discussion_r1105891068) as i cannot understand it

File: esm/index.js (hunk 1 above)
Comment (qwelias) [reply esm/index.js:73]: it was not inherited

Comment (WebReflection) [reply esm/index.js:73]: too many signals libraries ... you are right, I thought Signal extends ReactiveSignal but it's the other way around ... OK then, let's add peek to computed!

**Comment by ilibar-zpt** [comment]
@WebReflection did I miss anything?

**Comment by qwelias** [comment]
@WebReflection did I miss anything?

**Pull request merged**